test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/ComparatorTest.scala

```diff
@@ -40,4 +45,190 @@ class ComparatorTest {
+    assertTrue(cmp.compare(null, "a") < 0)
+    assertTrue(cmp.compare("a", null) > 0)
+  }
+
+  @Test def nullsFirstNull(): Unit = {
+    val cmp = ju.Comparator.nullsFirst(null)
+
+    assertEquals(0, cmp.compare("a", "b"))
+    assertEquals(0, cmp.compare(null, null))
+    assertTrue(cmp.compare(null, "a") < 0)
+    assertTrue(cmp.compare("a", null) > 0)
+  }
+
+  @Test def nullsLast(): Unit = {
+    val cmp = ju.Comparator.nullsLast(ju.Comparator.naturalOrder[String])
+    assertEquals(0, cmp.compare("a", "a"))
+    assertEquals(0, cmp.compare(null, null))
+    assertTrue(cmp.compare(null, "a") > 0)
+    assertTrue(cmp.compare("a", null) < 0)
+  }
+
+  @Test def nullsLastNull(): Unit = {
+    val cmp = ju.Comparator.nullsLast(null)
+    assertEquals(0, cmp.compare("a", "b"))
+    assertEquals(0, cmp.compare(null, null))
+    assertTrue(cmp.compare(null, "a") > 0)
+    assertTrue(cmp.compare("a", null) < 0)
+  }
+
+  @Test def comparing(): Unit = {
+    val cmp = ju.Comparator.comparing[String, String](
+        (_.substring(1)): juf.Function[String, String],
+        ju.Comparator.reverseOrder[String])
+    assertEquals(0, cmp.compare("ac", "bc"))
+    assertTrue(cmp.compare("ba", "ab") > 0)
+    assertTrue(cmp.compare("ab", "ba") < 0)
+
+    assertThrowsNPEIfCompliant(ju.Comparator.comparing[String, String](
+        null, ju.Comparator.reverseOrder[String]))
+    assertThrowsNPEIfCompliant(ju.Comparator.comparing(
+        (_.substring(1)): juf.Function[String, String], null))
+  }
+
+  @Test def comparingComparable(): Unit = {
+    val cmp = ju.Comparator.comparing[String, String](
+        (_.substring(1)): juf.Function[String, String])
+    assertEquals(0, cmp.compare("ac", "bc"))
+    assertTrue(cmp.compare("ba", "ab") < 0)
+    assertTrue(cmp.compare("ab", "ba") > 0)
+
+    assertThrowsNPEIfCompliant(
+        ju.Comparator.comparing[String, String](null))
+  }
+
+  @Test def comparingInt(): Unit = {
+    val cmp = ju.Comparator.comparingInt((_: String).length)
+    assertEquals(0, cmp.compare("a", "b"))
+    assertTrue(cmp.compare("", "a") < 0)
+    assertTrue(cmp.compare("ab", "") > 0)
+
+    assertThrowsNPEIfCompliant(ju.Comparator.comparingInt(null))
+  }
+
+  @Test def comparingLong(): Unit = {
+    val cmp = ju.Comparator.comparingLong((_: String).length.toLong)
+    assertEquals(0, cmp.compare("a", "b"))
+    assertTrue(cmp.compare("", "a") < 0)
+    assertTrue(cmp.compare("ab", "") > 0)
+
+    assertThrowsNPEIfCompliant(ju.Comparator.comparingLong(null))
+  }
+
+  @Test def comparingDouble(): Unit = {
+    val cmp = ju.Comparator.comparingDouble((_: String).length.toDouble / 2)
+    assertEquals(0, cmp.compare("a", "b"))
+    assertTrue(cmp.compare("", "a") < 0)
+    assertTrue(cmp.compare("ab", "") > 0)
+
+    assertThrowsNPEIfCompliant(ju.Comparator.comparingDouble(null))
+  }
+
+  @Test def thenComparingComparator(): Unit = {
+    val base = ju.Comparator.comparingInt((_: (Int, Int))._1)
```

Comment:
This is not super readable, IMO. Consider the following style when you have a tuple argument:
```suggestion
    val base = ju.Comparator.comparingInt((x: (Int, Int)) => x._1)
```